assets/questions/bstClosestValue.js

```diff
@@ -0,0 +1,173 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Question data for Find Closest Value in Binary Search Tree.
+ */
+/* eslint no-magic-numbers: ["error", {ignore: [1,4,14.7,2.5,24]}]*/
+globalData.questions['bstClosestValue'] = {  // eslint-disable-line dot-notation
+  title: 'Find Closest Value in Binary Search Tree',
+  starterCode: {
+    python:
+`# The tree is represented as a preorder traversal sequence of node values, 
+# in which '#' represents an empty node.
+# For example:
+# The input '2 1 # # #' represents the serialized format of a binary tree
+# using preorder order traversal, where root value is 2 and only has a left
+# child with value 1
+#
+# Definition for a binary tree node.
+# class TreeNode(object):
+#     def __init__(self, x):
+#         self.val = x
+#         self.left = None
+#         self.right = None
+
+def findClosestValue(input):
+    root = input[0]
+    target = input[1]
+
+    #TODO : write your code after here
+    return ""
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+        @classmethod
+        def deserialize(cls, data):
+            class TreeNode(object):
+                def __init__(self, x):
+                    self.val = x
+                    self.left = None
+                    self.right = None
+
+
+                def __str__(self):
+                    return self.val
+
+            def constructTree():
+                val = next(vals)
+                if val == '#':
+                    return None
+                node = TreeNode(float(val))
+                node.left = constructTree()
+                node.right = constructTree()
+                return node
+
+            inputs = data.split('|')
+            vals = iter(inputs[0].split())
+            return constructTree(), float(inputs[1])
+
+        @classmethod
+        def failToUseAbsoluteDiff(cls, input):
+            root = input[0]
+            target = input[1]
+
+            kid = root.left if target < root.val else root.right
+            if not kid:
+                return root.val
+            closest = cls.failToUseAbsoluteDiff([kid,target])
+            return root.val if (root.val - target) < (closest - target) else closest
+
+        @classmethod
+        def failToCompareAll(cls, input):
+            root = input[0]
+            target = input[1]
+
+            kid = root.left if target < root.val else root.right
+            if not kid:
+                return root.val
+            return cls.failToCompareAll([kid,target])
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'In this question, you need to find the closest value in a non-empty binary search tree ',
+        'given the target value. The target value is given in float format. It\'s guaranteed ',
+        'that there\'s only one closest value. You will have the root TreeNode. Each TreeNode ',
+        'has attributes val, left and right.'
```

Comment:
![screenshot from 2017-04-12 16 51 29](https://cloud.githubusercontent.com/assets/7578062/24984107/537d2e12-1fa0-11e7-9093-ab0558d0eabe.png)

The comments are not in the code, it's in webpage which users will see it. The reason why we don't put it in the instructions is that it's hard to put multiline strings in the instruction struct. It seems to strip the whitespace automatically.